### DISCUSSION TITLE
Component refreshing

### DISCUSSION THREAD

**Discussion by iabduul7** [livewire/livewire#7246] [1. Help]
Since Livewire 3, we cannot directly do `protected $listeners = ['refreshComponent' => '$refresh'];` so if I want to refresh the compnent using dispatch event like `$this->dispatch('refreshComponent');` and `
#[On('refreshComponent')]
    public function update()
    {
        //
    }
`

How Do I refresh the component here?

**Comment by edwinvdpol** [comment]
I would like to add, that it also isn't possible to listen to multiple events with `#[On()]`.

When you have for instance a create- and update action and the same function needs to be executed, you have to make 2 seperate methods for each, which do the same.

Maybe the possibility to add multiple events as an array is an option, like for example:
```php
#[On(['created', 'updated'])]
public function refreshSomething(): void {}
```

**EDIT:**
Ah, you can provide multiple attributes like this, my bad!
```php
#[On('created')]
#[On('updated')]
public function refreshSomething(): void {}
```

**Reply by iabduul7** [reply]
Oh cool, will defo use this.
